### DISCUSSION TITLE
Initial Lens Model for DESIJ1421-0022

### DISCUSSION THREAD

**Discussion by SamWGamgee** [AstroBridge/BDLensing#152] [Q&A] [answered]
Using this mask, I achieved chi^2=0.88. 
![image](https://github.com/ajshajib/BDLensing/assets/90761471/0d36c575-6c23-47d0-8645-18a15d44a86c)

But there's a lot of light in source light and source light convolved.
![image](https://github.com/ajshajib/BDLensing/assets/90761471/b458243d-f572-478b-b9b3-504f2a12965d)
![image](https://github.com/ajshajib/BDLensing/assets/90761471/c7b84394-26bd-4e73-b211-fa2ace742466)

I actually tried unmasking blobs at 11 and 5 o'clock as these seemed part of the arc. But that didn't give a decent chi^2 (2.34).
![image](https://github.com/ajshajib/BDLensing/assets/90761471/734884e3-af6f-478a-8457-c99593ddd3cd)
![image](https://github.com/ajshajib/BDLensing/assets/90761471/22e93a9c-7d8a-4c76-9bd1-43314161d231)

And it's not fixing the extra light problem in source light plot. 

So what to do in this regard? Any suggestions?

**Comment by aaimtiaz** [comment]
Can you provide the values of the parameter from the PSO output?

**Reply by SamWGamgee** [reply]
Sure.

First one: (chi^2=0.88)
{'kwargs_lens': [{'theta_E': 0.96297290985463, 'gamma': 1.764005623682797, 'e1': 0.15832623189252157, 'e2': -0.11454437955693268, 'center_x': 0.18850131236016593, 'center_y': -0.48178931401446456}, {'gamma1': 0.040740487674490955, 'gamma2': -0.2041793257491808, 'ra_0': 0, 'dec_0': 0}], 

'kwargs_source': [{'amp': 5.478741600912629, 'R_sersic': 0.188232606469556, 'n_sersic': 1.0, 'e1': -0.11589505582545957, 'e2': -0.19496018072628618, 'center_x': 0.40078773508494336, 'center_y': -0.012061707152879459}, {'amp': array([ 50.24975915,  42.40306313,  -0.40079495,  64.10533459,
        15.59412403,  61.91787219,  45.1040275 ,   2.40946491,
        47.88381928, -15.66708505,  44.11355757,   2.26862515,
        44.87948214,   9.48460352,  26.84770276,  -7.1218326 ,
         3.92333481,   1.46322415,  -0.94766283,   2.67320067,
       -12.86470957,   8.63990601,   7.69358234,  -7.75563587,
        -3.82475168,  13.99523828,  -0.48304558,  11.55120428,
       -38.07593727,   0.20310582, -15.02704   , -13.47772194,
        -9.29559726,  -6.85074577,  -8.17045819,  -1.58365769,
        13.67060118,  -6.72439081,   1.20730924,  -1.61086798,
        -3.49344007,  -0.39075179,   5.74994711,   0.45079498,
        -1.09583831, -10.76276761,   0.41144802,   0.81962655,
        -3.62353633,  -9.21802358,  -2.22689072,  -5.68095957,
         6.79890521,  -2.80104102,   1.16981328,  27.40319883,
        -6.47093345,   7.74083606,   8.73541693,  10.56725093,
         6.00238371,   8.7850937 ,   4.15471197,   4.38409706,
        -4.56028089,   4.1783887 ]), 'n_max': 10, 'beta': 0.4673896622012622, 'center_x': 0.40078773508494336, 'center_y': -0.012061707152879459}], 

'kwargs_lens_light': [{'amp': 159.52367279034092, 'R_sersic': 0.49657827416794215, 'n_sersic': 4.0, 'e1': 0.15832623189252157, 'e2': -0.11454437955693268, 'center_x': 0.18850131236016593, 'center_y': -0.48178931401446456}], 'kwargs_ps': [], 'kwargs_special': {}, 'kwargs_extinction': []}

**Reply by SamWGamgee** [reply]
2nd one: (chi^2=2.34)

{'kwargs_lens': [{'theta_E': 1.0723218211449914, 'gamma': 1.381979720214526, 'e1': 0.1828180112125753, 'e2': -0.1409794695191765, 'center_x': 0.18676215157802567, 'center_y': -0.482520777816265}, {'gamma1': 0.08003923353047565, 'gamma2': -0.11330238370183073, 'ra_0': 0, 'dec_0': 0}], 

'kwargs_source': [{'amp': 24.883467425783024, 'R_sersic': 0.20274433010131276, 'n_sersic': 1.0, 'e1': 0.24492319232712081, 'e2': 0.0502358812814978, 'center_x': 0.2736347832017433, 'center_y': -0.12084637562683048}, {'amp': array([ 60.52419227,  89.74585096,  99.38481292,  47.1051307 ,
        75.04396439, 117.61427516,   9.87940349, -41.54186606,
        46.62818725, -18.1067097 ,  33.88514345, -84.14161008,
        -3.39753481,   1.75755912,  -8.69715852,  30.59851485,
       -41.99691781, -33.16713054, -34.81361273, -19.74569685,
       -60.55827776,  14.96287046,   8.58791126,  22.02101876,
       -15.09075244,  49.77883611, -12.9826879 ,  36.22114141,
       -30.43716349,  -3.17346277,  35.12084157, -12.066409  ,
        24.69374509, -20.12581884,  -2.08393184,   3.12123664,
        -5.13441477,  -9.69536478,  23.65576541,  -1.49100095,
        16.72832301,   1.72778066,  24.24389654,  11.17525327,
        12.26619399,  -5.63466335,  -6.2829218 , -13.64819888,
        -2.44783051,  -6.63183469, -11.79355077,  -0.30932051,
       -18.16464218,  13.16682137,  -9.10538453,  19.06551508,
        21.04983486, -10.50814579,  -0.37982763,   3.07687561,
       -17.63696386,  13.10838836, -14.90519234,   5.9119864 ,
       -16.41949138,  -6.48681729]), 'n_max': 10, 'beta': 0.24487526756699068, 'center_x': 0.2736347832017433, 'center_y': -0.12084637562683048}], 

'kwargs_lens_light': [{'amp': 142.693678643632, 'R_sersic': 0.5296494797638817, 'n_sersic': 4.0, 'e1': 0.1828180112125753, 'e2': -0.1409794695191765, 'center_x': 0.18676215157802567, 'center_y': -0.482520777816265}], 'kwargs_ps': [], 'kwargs_special': {}, 'kwargs_extinction': []}

**Reply by aaimtiaz** [reply, marked as answer]
These are the things I would try:

1. Setting the lower value of theta_E bit higher, guessing from the figure.
2. Setting 0.1 as upper value of beta
3. Fixing  the gamma value around 1.8 or 2. You can also try different values.
4. Furthermore, adding a second sersic profile to the lens light list, if the above steps do not work.

**Reply by rafee-adnan** [reply]
- Make the outer mask bigger.
- Remove the mask at 1'o clock and 5'o clock, and rerun the pso. See if you get better fit in those areas.
- Follow this discussion #130, you are facing similar problem

**Reply by SamWGamgee** [reply]
I actually got a chi^2=1.328 after following @aaimtiaz vai's suggestions and @rafee-adnan vai's point 1 and also some other directions from him. I have some other ideas to lower the pso but just wanted to make sure if the source light plots are valid.

Here's the output:
{'kwargs_lens': [{'theta_E': 2.1294604489067632, 'gamma': 2.0, 'e1': 0.13297498298158503, 'e2': 0.26805542557698153, 'center_x': 0.3354338002211309, 'center_y': -1.0592942064981254}, {'gamma1': -0.044538163323028034, 'gamma2': 0.07764931669798299, 'ra_0': 0, 'dec_0': 0}], 'kwargs_source': [{'amp': 104.71075092658172, 'R_sersic': 0.1998221236496937, 'n_sersic': 1.0, 'e1': -0.03530557522161587, 'e2': -0.499160856391953, 'center_x': 0.6909868499035541, 'center_y': -0.5308121695879726}, {'amp': array([-155.89154874,  243.38982008,  295.16276849,   84.53805949,
        327.79892846,  398.57181016,   35.56779615,  -84.62158989,
        127.74488295,  247.53473734,   -2.96202744,  -18.1226035 ,
       -183.61456017,   65.39628615,  133.57304688,   -2.41914485,
         13.53700748,    9.08443308, -102.49998742,   39.04233088,
         32.45524767,    6.37608483,   -5.44177986,   91.08157007,
         46.38791999,  -37.86097233,    6.53905755,   28.47431092,
         14.12176804,  -72.34057238,   27.99897777,   48.23096793,
         21.84302227,   -8.78685217,    3.87228866,    6.59366366,
         16.53010387,  -12.48381566,  -62.99136313,   -4.47870956,
         -2.60412884,   -0.47143462,   -5.45472889,  -11.68055293,
         62.58484013,   -8.16896094,   29.06136121,   16.09300008,
        -67.86313823,   29.78867049,  -47.98301522,   20.15531283,
        -29.13479461,   11.67085349,   54.86982194,   -1.9531111 ,
        -13.63981836,   45.66509997,  -19.76096359,   -1.36304239,
          3.78970421,   -7.0679556 ,   -1.54108994,   -4.34203332,
          5.64393813,   32.96550738]), 'n_max': 10, 'beta': 0.13042558813744998, 'center_x': 0.6909868499035541, 'center_y': -0.5308121695879726}], 'kwargs_lens_light': [{'amp': 112.4592626651367, 'R_sersic': 0.7367746255829111, 'n_sersic': 4.0, 'e1': 0.13814485977133464, 'e2': -0.20867836499606443, 'center_x': 0.18922580502547637, 'center_y': -0.4814474443717736}, {'amp': -75.95211435894905, 'R_sersic': 0.4578919721386282, 'n_sersic': 1.0, 'e1': 0.11117701099463914, 'e2': -0.4178341116747966, 'center_x': 0.18922580502547637, 'center_y': -0.4814474443717736}], 'kwargs_ps': [], 'kwargs_special': {}, 'kwargs_extinction': []}

![image](https://github.com/ajshajib/BDLensing/assets/90761471/c533b961-acc2-403d-8a2a-7f4848fd2619)
![image](https://github.com/ajshajib/BDLensing/assets/90761471/49ea691e-59de-45c4-b37c-9343dd0b6a0d)

**Reply by rafee-adnan** [reply]
Would you mind sharing your latest notebook? @SamWGamgee

**Reply by SamWGamgee** [reply]
Here it is, @rafee-adnan vai
https://www.kaggle.com/code/masukridwan/rafee-adnan-vai?scriptVersionId=135500802
I actually named two separate files on you and A I Imtiaz vai :3 
Can you see the file?

**Reply by rafee-adnan** [reply]
Yes, I can.

**Reply by aaimtiaz** [reply]
Have you tried modeling without masking the light regions at 2 o'clock and 5 o'clock?

**Reply by SamWGamgee** [reply]
@aaimtiaz vai, yes. Here's the plots of that version - 

![image](https://github.com/ajshajib/BDLensing/assets/90761471/5ecac257-47d7-4387-a2bf-11d0a22880c1)
![image](https://github.com/ajshajib/BDLensing/assets/90761471/325f9eeb-bbc2-4365-96ef-d4120eb27b8e)

Results: 
{'kwargs_lens': [{'theta_E': 2.1802210112678853, 'gamma': 2.003294823370444, 'e1': -0.09235783334740998, 'e2': 0.2069224537946741, 'center_x': 0.43310443624908124, 'center_y': -1.198907644282366}, {'gamma1': -0.13250495004215115, 'gamma2': 0.014654444659801666, 'ra_0': 0, 'dec_0': 0}], 'kwargs_source': [{'amp': 26.689848890945008, 'R_sersic': 0.054162912855898734, 'n_sersic': 1.0, 'e1': 0.0798654710178168, 'e2': -0.1858816904221295, 'center_x': 0.7771690910226894, 'center_y': -0.526309761929768}, {'amp': array([204.15964275, 283.64994369, 161.60457064, 214.36348052,
       172.84448242, 234.4340539 ,  95.71802458,  76.36171389,
       232.27217996,  97.74512433,  32.3705116 , -19.86473166,
       119.75645489, 114.58965741, 110.06241584,  45.24572416,
       -24.11076223,  20.87842767,  50.62430678,  92.10301965,
        13.1089339 ,  33.1877346 ,  13.89535194,  16.76636761,
         3.55636909,  88.17711657,  34.7503236 ,  61.60185268,
        15.4251848 ,   5.50649677,  45.96695206, -24.23427232,
        24.52058478,   5.66923932,  19.98889754,  20.54174215,
        -6.32245542, -36.21672552,  62.63167001,   7.51972775,
        21.36273375,   6.3249143 ,  -5.05877732,   3.55129917,
        32.30342246,  18.02521964, -51.25878623,   0.63351585,
        -8.17629306,   6.36759406,   5.15278896,   0.4383548 ,
        30.43927349,  51.84971613,  40.87340049,  17.33701421,
        -6.3444303 ,  -9.64020492, -21.05606864, -11.49370484,
       -20.9336667 , -13.03526838, -16.44935304,  31.94212819,
        12.93030939,  36.53997214]), 'n_max': 10, 'beta': 0.11452009733956273, 'center_x': 0.7771690910226894, 'center_y': -0.526309761929768}], 'kwargs_lens_light': [{'amp': 78.29746444266276, 'R_sersic': 1.0509132242829906, 'n_sersic': 4.0, 'e1': 0.11501578038383928, 'e2': -0.3319421098274766, 'center_x': 0.18899840000195184, 'center_y': -0.48294382483419845}, {'amp': -105.0769125994361, 'R_sersic': 0.606669604156648, 'n_sersic': 1.0, 'e1': 0.07719964226571333, 'e2': -0.49157554750012045, 'center_x': 0.18899840000195184, 'center_y': -0.48294382483419845}], 'kwargs_ps': [], 'kwargs_special': {}, 'kwargs_extinction': []}
